### ISSUE TITLE
Keybase is not launching/starting on Linux Ubuntu

### ISSUE THREAD

**Issue by santiagocasas** [keybase/client#23073]
Keybase is not launching/starting on Linux Ubuntu 
Either launching from terminal or from the Startup Menu, the window appears some milisecods and disappears immediatly.
Reinstall already performed, using instructions on the site. And also using snap.

my log id: d72943d67019278d7da3481c

**Comment by ricardobalk** [comment]
Hi @santiagocasas, I'm having the same issue, after upgrading my package version.

- The CLI works
- kbfs is accessible (yay!)
- Once I click on the launcher button (the one from `/usr/share/applications/keybase.desktop`), it shows a window and closes it.

I'm using Ubuntu 18.04.4 LTS, `uname -a` gives me `Linux neptune 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

Tried to clean kill Keybase with `run_keybase -k`, clean the Keybase cache directory and I tried to reinstall Keybase via `apt`. I wish I could do something similar to `sudo apt install keybase:amd64=5.2.0-20200130211428.cf82db8320 -V` to install an older version... but that does not work.

I also tried to execute `/opt/keybase/Keybase`, but that gave me:
```
undisclosed@neptune:~$ /opt/keybase/Keybase
Version: 5.3.0-20200310205642+4f2689009b
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setAutoHideMenuBar function' is deprecated and will be removed. Please use 'autoHideMenuBar property' instead.
Segmentation fault (core dumped)
undisclosed@neptune:~$ 
```

**Comment by major0** [comment]
Running into the exact same problem on Ubuntu 19.10:
```
/opt/keybase/Keybase -k
Version: 5.3.0-20200310205642+4f2689009b
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setAutoHideMenuBar function' is deprecated and will be removed. Please use 'autoHideMenuBar property' instead.
Segmentation fault (core dumped)
```

**Comment by major0** [comment]
dmesg output from the segfault:
```
segfault at 73006e ip 00005569908dbc9b sp 00007fff10229800 error 4 in Keybase[55698b7b1000+53d3000]
```

**Comment by heronhaye** [comment]
Thanks, we're looking into it. Could everyone mention what DE/WM they're using?

**Comment by Bayernatoor** [comment]
Hey, experiencing this as well Ubuntu 19.10 - @heronhaye  What does DE/WM stand for?

**Comment by major0** [comment]
Default Ubuntu 19.10 Gnome/GTK desktop.

```
$ echo $DESKTOP_SESSION
ubuntu
$ echo $XDG_CURRENT_DESKTOP
ubuntu:GNOME
```

**Comment by heronhaye** [comment]
Thanks. Would anyone be willing to build the Keybase GUI locally to see if a potential fix works? (I cannot reproduce this issue myself.)

**Comment by ricardobalk** [comment]
Sure, I have Golang installed so that shouldn't be a problem :)

**Comment by heronhaye** [comment]
Needs yarn for the GUI, actually. 

1) Clone https://github.com/keybase/client/.
2) `cd shared`
3) `yarn modules` 
4) `yarn start`

This should build the electron app from master against your running keybase service. Let me know if it crashes (it should, otherwise let me know).

Then try checking out `surya/menubar-segfault` (https://github.com/keybase/client/pull/23064) and rerunning `yarn start`. Hopefully this shouldn't crash.

**Comment by ricardobalk** [comment]
Ah, yeah, `yarn` makes sense... Go for the backend and yarn for the GUI

The default branch crashed, the same behaviour as with the pre-compiled Keybase client.

After a `git checkout surya/menubar-segfault` and `yarn modules && yarn start` the GUI showed up.

![image](https://user-images.githubusercontent.com/14904229/76782654-e7422c00-67b0-11ea-8181-50ad44cc751a.png)

**Comment by heronhaye** [comment]
Wow, that's really great news since I was kind of flying blind without a repro. I'll try to get this merged and push out a release.

Does everything in the GUI look okay?

(If anyone else in this thread wants to give it a shot, that'd be appreciated too!)

**Comment by ricardobalk** [comment]
Yup, everything works great as far as I could see. :+1:

**Comment by ricardobalk** [comment]
This was merged in commit a73fc22cd78598792f12ac1337b005b71845bbb5, right? And the separate branch got deleted?

**Comment by ricardobalk** [comment]
So people could now just do

> 1) Clone https://github.com/keybase/client/.
> 2) `cd shared`
> 3) `yarn modules` 
> 4) `yarn start`

without checking out `surya/menubar-segfault` (which does not exist any longer)

or wait for the release.


Wrote this for those that came here from my references

**Comment by heronhaye** [comment]
Sorry, we're having some issues with the release process currently. Hoping to get it out soon. But yes, that should work.

**Comment by heronhaye** [comment]
If you like, you could try out a test build at https://s3.amazonaws.com/tests.keybase.io/linux_binaries/deb/keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb. Signature is at https://s3.amazonaws.com/tests.keybase.io/linux_binaries/deb/keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb.sig, and you can verify with our code signing key at https://keybase.io/docs/server_security/our_code_signing_key.

**Comment by heronhaye** [comment]
@santiagocasas @major0 @ricardobalk @Bayernatoor please try out the nightly build (instructions at https://keybase.io/docs/linux-user-guide#nightly-builds) and let me know if it fixes the issue for you. I'll update when the stable release is out.

**Comment by daniel156161** [comment]
@heronhaye nightly build working fine and icon is not broken

**Comment by ricardobalk** [comment]
Did the following:

```sh
curl -s https://keybase.io/docs/server_security/code_signing_key.asc | gpg --import
cd ~/Downloads
wget https://s3.amazonaws.com/tests.keybase.io/linux_binaries/deb/keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb
wget https://s3.amazonaws.com/tests.keybase.io/linux_binaries/deb/keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb.sig
gpg --verify keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb.sig
sudo dpkg -i keybase_5.4.0-20200319210509.6b357fbc3f_amd64.deb
run_keybase -k
run_keybase
```

And it works! :smile: 

![image](https://user-images.githubusercontent.com/14904229/77170811-e1539000-6abb-11ea-8106-1a29abe72206.png)

**Comment by heronhaye** [comment]
All right, we've released version 5.3.1 which should fix the issue. Please upgrade and let me know if it still doesn't work.

If you installed the nightly, you can downgrade to the release version (manually reinstall from https://keybase.io/docs/the_app/install_linux), or you can just stay on the nightly (you'll get an update for the next stable release when it comes out).

**Comment by rdkvx** [comment]
> 
I tried this and worked, thx mate.
